### ISSUE TITLE
Pylance reports a problem when a walrus operator is used in a class member tuple/list

### ISSUE THREAD

**Issue by dhyxy** [microsoft/pylance-release#1213]
## Environment data

-   Language Server version: v2021.4.3
-   OS and version: macOS Big Sur 11.2.3
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.9.1 (pyenv)

## Expected behaviour

When a walrus operator is used in a tuple/list as a class variable, the syntax should be considered valid. (included code below)

output from run:
<img width="672" alt="image" src="https://user-images.githubusercontent.com/58804183/116640628-436dfe00-a928-11eb-9632-0a67de5bd8e7.png">

## Actual behaviour

Pylance marks it as a problem. This only occurs when it's a class variable, it doesn't show any error if it's in an `__init__` method. Specifically:
```json
{
	"resource": "/Users/dhyey/Projects/temp/pylance_walrus/a.py",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 8,
	"message": "Assignment expression must be within module, function or lambda",
	"source": "Pylance",
	"startLineNumber": 3,
	"startColumn": 10,
	"endLineNumber": 3,
	"endColumn": 20
}
```

<img width="317" alt="image" src="https://user-images.githubusercontent.com/58804183/116640226-a8752400-a927-11eb-9697-0ff20073e3b3.png">

<img width="616" alt="image" src="https://user-images.githubusercontent.com/58804183/116639851-b5454800-a926-11eb-8f7a-c521acfa4af6.png">

## Logs

couldn't figure out how to see the pylance logs 😅 but I found this
```
[2021-04-29 20:02:04.399] [exthost] [warning] ms-python.python - Code actions of kind 'quickfix 'requested but returned code action is of kind 'refactor.extract'. Code action will be dropped. Please check 'CodeActionContext.only' to only return requested code actions.
```

## Code Snippet / Additional information

```python
class A:
    ROLES = (
        (ROLE1 := 1, "role1"),
        (ROLE2 := 2, "role2"),
    )

print(A.ROLE1)
print(A().ROLE1)
print(A.ROLE2)
print(A.ROLES)
```

you can bypass this by instantiating the specific variables outside of the tuple but I thought it might be worth bringing up, thank you in advance 🥳

**Comment by erictraut** [comment]
Thanks for the bug report. 

I misinterpreted a statement in [this section of PEP 572](https://www.python.org/dev/peps/pep-0572/#scope-of-the-target) to mean that assignment expressions were disallowed within a class scope. But after rereading it, it's now clear that it's specifically referring to assignment expressions within comprehensions within a class scope.

This is an easy fix, and it will be address in an upcoming release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.5.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202150-5-may-2021
